bundles/org.openhab.binding.plugwiseha/src/main/java/org/openhab/binding/plugwiseha/internal/handler/PlugwiseHAApplianceHandler.java

```diff
@@ -0,0 +1,426 @@
+                }
+                break;
+            case APPLIANCE_LOCK_CHANNEL:
+                Boolean relayLockState = entity.getRelayLockState().orElse(null);
+                if (relayLockState != null) {
+                    state = OnOffType.from(relayLockState);
+                }
+                break;
+            case APPLIANCE_OFFSET_CHANNEL:
+                if (entity.getOffsetTemperature().isPresent()) {
+                    Unit<Temperature> unit = entity.getOffsetTemperatureUnit().orElse(UNIT_CELSIUS).equals(UNIT_CELSIUS)
+                            ? SIUnits.CELSIUS
+                            : ImperialUnits.FAHRENHEIT;
+                    state = new QuantityType<Temperature>(entity.getOffsetTemperature().get(), unit);
+                }
+                break;
+            case APPLIANCE_POWER_CHANNEL:
+                if (entity.getRelayState().isPresent()) {
+                    state = OnOffType.from(entity.getRelayState().get());
+                }
+                break;
+            case APPLIANCE_POWER_USAGE_CHANNEL:
+                if (entity.getPowerUsage().isPresent()) {
+                    state = new QuantityType<Power>(entity.getPowerUsage().get(), Units.WATT);
+                }
+                break;
+            case APPLIANCE_SETPOINT_CHANNEL:
+                if (entity.getSetpointTemperature().isPresent()) {
+                    Unit<Temperature> unit = entity.getSetpointTemperatureUnit().orElse(UNIT_CELSIUS)
+                            .equals(UNIT_CELSIUS) ? SIUnits.CELSIUS : ImperialUnits.FAHRENHEIT;
+                    state = new QuantityType<Temperature>(entity.getSetpointTemperature().get(), unit);
+                }
+                break;
+            case APPLIANCE_TEMPERATURE_CHANNEL:
+                if (entity.getTemperature().isPresent()) {
+                    Unit<Temperature> unit = entity.getTemperatureUnit().orElse(UNIT_CELSIUS).equals(UNIT_CELSIUS)
+                            ? SIUnits.CELSIUS
+                            : ImperialUnits.FAHRENHEIT;
+                    state = new QuantityType<Temperature>(entity.getTemperature().get(), unit);
+                }
+                break;
+            case APPLIANCE_VALVEPOSITION_CHANNEL:
+                if (entity.getValvePosition().isPresent()) {
+                    state = new QuantityType<Dimensionless>((int) (entity.getValvePosition().get() * 100),
+                            Units.PERCENT);
+                }
+                break;
+            case APPLIANCE_WATERPRESSURE_CHANNEL:
+                if (entity.getWaterPressure().isPresent()) {
+                    Unit<Pressure> unit = HECTO(SIUnits.PASCAL);
+                    state = new QuantityType<Pressure>(entity.getWaterPressure().get(), unit);
+                }
+                break;
+            default:
+                break;
+        }
+
+        if (state != UnDefType.NULL && state != UnDefType.UNDEF) {
```

Comment:
If the state for the channel can't be set it does make sense to set to UNDEF. So not sure why it's excluded here?